### DISCUSSION TITLE
Publish Docker Image

### DISCUSSION THREAD

**Discussion by dphildebrandt** [lukevella/rallly#268] [Ideas]
Any thoughts on publishing an official image, so we could just consume and add to existing docker-compose files to make it easier to add to existing self-hosted stacks?

**Comment by xiNeRoar** [comment]
strongly agree with this. Please have a second though of publishing a public image to make things easier

**Comment by justinmayer** [comment]
Now that @LarsHaalck has graciously fixed #170 by removing the `DATABASE_URL` variable from the Dockerfile in #328, I wonder if we could re-visit the topic of [publishing Docker images](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images) as part of Rallly’s GitHub Action-powered CI/CD process. As I understand from the discussion around the time the [previous CD configuration was removed](https://github.com/lukevella/rallly/commit/9098bba37cd7739ea59ae51d17f52e95ca25d5cd), the `DATABASE_URL` variable’s presence in the Dockerfile was the primary blocker.

With that blocker removed, what (if anything) remains to be done before resuming Docker image auto-publication?

What do you think, @lukevella?

**Reply by lukevella** [reply]
I don't believe there's anything blocking this now. I wasn't planning on working on this straight away as I'm in the middle of updates to the actual app so unless this is more urgent than I understand it to be it might be sometime before I can get to it.

**Reply by dphildebrandt** [reply]
I think this would greatly boost adoption of this project.

**Reply by lukevella** [reply]
I hear you, but my current priority is the next major release and I can't spare the time to work on this. If someone else wants to help out on this front I'd welcome a PR.

**Reply by samsapti** [reply]
#349

**Comment by brettinternet** [comment]
RE: #349 The docker username is a secret? `secrets.DOCKER_USERNAME`

Can we link the official docker registry somewhere? I'd also like to add a couple steps to add multi-arch support if there's interest.

**Reply by justinmayer** [reply]
I think you are correct that the Docker username need not be a secret.

I believe the resulting Docker image is published to Docker Hub: https://hub.docker.com/r/lukevella/rallly

**Reply by lukevella** [reply]
Fair point, probably doesn't need to be a secret but it helps to have it be in some sort of stored value as it needs to be consistent between the login command and the push command. @justinmayer is right, that is the correct docker hub repository. I think now that there is an image it would make sense to create a dedicated repo for self-hosting with docker since it's not necessary to check out the entire codebase to get it running anymore and that repo can link to the docker hub repo.
